src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java

```diff
@@ -965,40 +975,102 @@ private void downloadFileAnnotation(RoutingContext event) {
-                            response.end("File for given Annotation does not exist");
-                            return;
+            ImageRegionVerticle.GET_FILE_ANNOTATION_METADATA_EVENT,
+            Json.encode(annotationCtx), new Handler<AsyncResult<Message<JsonObject>>>() {
+                @Override
+                public void handle(AsyncResult<Message<JsonObject>> result) {
+                    if (result.failed()) {
+                        log.error(result.cause().getMessage());
+                        response.setStatusCode(404);
+                        response.end("Could not get annotation "
+                                    + request.getParam("annotationId"));
+                        return;
+                    }
+                    JsonObject fileInfo = result.result().body();
+                    String fileName = fileInfo.getString("originalFileName");
+                    String filePath = fileInfo.getString("originalFilePath");
+                    //If the path is a directory, send error response
+                    File file = new File(filePath);
+                    if (!file.exists()) {
+                        response.setStatusCode(404);
+                        response.end("File for given Annotation does not exist");
+                        return;
+                    }
+                    if (file.isDirectory()) {
+                        response.setStatusCode(501);
+                        response.end("File Annotation of Unsupported File Type");
+                        return;
+                    }
+                    else {
+                        String contentType = "application/octet-stream";
+                        try {
+                            contentType = new Tika().detect(file);
+                        } catch (IOException e) {
+                            log.warn(String.format(
+                                    "Failed to detect content type of file %s",
+                                    fileName));
                         }
-                        if (file.isDirectory()) {
-                            response.setStatusCode(501);
-                            response.end("File Annotation of Unsupported File Type");
+                        response.headers().set("Content-Type", contentType);
+
+                        response.headers().set("Accept-Ranges", "bytes");
+                        ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(
+                                Instant.ofEpochMilli(file.lastModified()),
+                                ZoneId.of("UTC"));
+                        DateTimeFormatter formatter =
+                                DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+                        response.headers().set("Last-Modified", zonedDateTime.format(formatter));
+                        response.headers().set("Content-Disposition",
+                                "attachment; filename=\"" + fileName + "\"");
```

Comment:
I made the mistake of looking up how filenames potentially containing non-ASCII characters should be encoded in HTTP headers: https://www.rfc-editor.org/rfc/rfc8187.html - I'm ok with just assuming filenames are sane, we don't use the filename from the header, and even if we did it wouldn't really matter what name the browser saves the file under. Perhaps there is a library that does all the work properly though?